Routing

AlekseiKolesnikov opened the pull request:
1. The routing was made by using state manager Zustand
2. Highlights of the chosen option are made as well

AlekseiKolesnikov committed:
1. The routing was made by using state manager Zustand
2. Highlights of chosen option is made as well

ShadowZilot commented on src/app/main.tsx:
```diff
@@ -1,10 +1,10 @@
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
+    <BrowserRouter>
+        <App/>
+    </BrowserRouter>,
```
Зачем убрал StrictMode, мне кажется лучше оставить

AlekseiKolesnikov committed:
1. Added Strict Mode back
2. Change the logic of finding the selected element, so now it doesn't matter what index in array it has

AlekseiKolesnikov committed:
1. The AppCategoryOptions.tsx were implemented with selection animations and navigation components. Later the API will be called with different parameters depends on the selected option.

ShadowZilot reviewed (approved):
Approve, but I've noticed some strange CSS classes names

AlekseiKolesnikov committed:
1. Change windowSizeState.ts logic, now it track the size of the window consistently not only after refresh
2. Put Routes in the single object
3. Fixed the bug with resizing, colors in AppNavbar.tsx and header_menu_icon

ShadowZilot commented on src/app/App.tsx:
```diff
@@ -1,11 +1,44 @@
+                {ROUTES.map((route) => (
+                    <Route key={route.path} path={route.path} element={route.element}/>
+                ))}
+            </Routes>
```
What if you need to add navigation within a section? Will you write Routes inside each section?

AlekseiKolesnikov committed:
Fixed ListOptionsContainer.tsx problem. Now options container scrolls whenever screen is too tiny.

ShadowZilot commented on src/components/layout/AppContent/AppContent.tsx:
```diff
@@ -1,10 +1,27 @@
+    return (
+        <div className={`content-container ${isFullScreenContent ? "notFullScreenWidth" : "fullScreenWidth"}`}>
+            <div>
+                {windowPathName !== '/' &&
```
Better use https://github.com/JedWatson/classnames for refer to class names from CSS

ShadowZilot commented on src/pages/TvShowsPage/TvShowsPage.tsx:
```diff
@@ -0,0 +1,21 @@
+
+    useEffect(() => {
+        setPathName(location.pathname)
+    }, [location.pathname, setPathName])
```
these lines are repeated in several components. It's better to put them in separate custom hook

AlekseiKolesnikov committed:
1. Fixed class names, now they dependent to each other
2. Fixed mobile version category options container bag, it was laying over sider

AlekseiKolesnikov committed:
1. Hided overflow in mobile screen

AlekseiKolesnikov committed:
1. Deleted unnecessary elements
